[Bug] Redis is throwing socket exception after staying idle for few minutes

## Description
```
Execution failed with status undefined
java.net.SocketException: Broken pipe (Write failed)
```

```
redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketException: Broken pipe (Write failed)
	at redis.clients.jedis.Connection.flush(Connection.java:268)
	at redis.clients.jedis.Connection.getOne(Connection.java:255)
	at redis.clients.jedis.Jedis.sendCommand(Jedis.java:3984)
	at com.external.plugins.RedisPlugin$RedisPluginExecutor.lambda$execute$0(RedisPlugin.java:82)
	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:91)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:126)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4090)
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketException: Broken pipe (Write failed)
	at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)
	at java.base/java.net.SocketOutputStream.socketWrite(Unknown Source)
	at java.base/java.net.SocketOutputStream.write(Unknown Source)
	at redis.clients.jedis.util.RedisOutputStream.flushBuffer(RedisOutputStream.java:52)
	at redis.clients.jedis.util.RedisOutputStream.flush(RedisOutputStream.java:133)
	at redis.clients.jedis.Connection.flush(Connection.java:265)
	... 15 more

```

--- a/app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java
+++ b/app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java
@@ -110,8 +110,8 @@ public class RedisPlugin extends BasePlugin {
                         /*
                          * - For some reason, Jedis throws a socket error when kept idle for like 10 min when
                          * appsmith is setup via docker image.
-                         * - APMU, jedis.close() should disconnect the connection, causing jedis to re-connect during
-                         *  next execution.
+                         * - APMU, jedis.close() should disconnect the connection, causing jedis to refresh connection
+                         * during next execution.
                          * - This is a placeholder solution till better fix is available (would connection pool fix
                          * it ?)
                          */

--- a/app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java
+++ b/app/server/appsmith-plugins/redisPlugin/src/main/java/com/external/plugins/RedisPlugin.java
@@ -36,7 +36,7 @@ import java.util.stream.Collectors;
 import static com.appsmith.external.constants.ActionConstants.ACTION_CONFIGURATION_BODY;
 
 public class RedisPlugin extends BasePlugin {
-    private static final Integer DEFAULT_PORT = 6379;
+    private static final Long DEFAULT_PORT = 6379L;
 
     public RedisPlugin(PluginWrapper wrapper) {
         super(wrapper);
@@ -105,6 +105,17 @@ public class RedisPlugin extends BasePlugin {
                         ActionExecutionResult result = actionExecutionResult;
                         result.setRequest(request);
                         return result;
+                    })
+                    .doFinally(signalType -> {
+                        /*
+                         * - For some reason, Jedis throws a socket error when kept idle for like 10 min when
+                         * appsmith is setup via docker image.
+                         * - APMU, jedis.close() should disconnect the connection, causing jedis to re-connect during
+                         *  next execution.
+                         * - This is a placeholder solution till better fix is available (would connection pool fix
+                         * it ?)
+                         */
+                        jedis.close();
                     })
                     .subscribeOn(scheduler);
         }
@@ -177,9 +188,6 @@ public class RedisPlugin extends BasePlugin {
                 Endpoint endpoint = datasourceConfiguration.getEndpoints().get(0);
                 if (StringUtils.isNullOrEmpty(endpoint.getHost())) {
                     invalids.add("Missing host for endpoint");
-                }
-                if (endpoint.getPort() == null) {
-                    invalids.add("Missing port for endpoint");
                 }
             }
 

--- a/app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java
+++ b/app/server/appsmith-plugins/redisPlugin/src/test/java/com/external/plugins/RedisPluginTest.java
@@ -83,7 +83,7 @@ public class RedisPluginTest {
         Endpoint endpoint = new Endpoint();
         invalidDatasourceConfiguration.setEndpoints(Collections.singletonList(endpoint));
 
-        Assert.assertEquals(Set.of("Missing port for endpoint", "Missing host for endpoint"),
+        Assert.assertEquals(Set.of("Missing host for endpoint"),
                 pluginExecutor.validateDatasource(invalidDatasourceConfiguration));
     }
 
@@ -95,7 +95,8 @@ public class RedisPluginTest {
         endpoint.setHost("test-host");
         invalidDatasourceConfiguration.setEndpoints(Collections.singletonList(endpoint));
 
-        Assert.assertEquals(pluginExecutor.validateDatasource(invalidDatasourceConfiguration), Set.of("Missing port for endpoint"));
+        // Since default port is picked, set of invalids should be empty.
+        Assert.assertEquals(pluginExecutor.validateDatasource(invalidDatasourceConfiguration), Set.of());
     }
 
     @Test
@@ -112,7 +113,7 @@ public class RedisPluginTest {
         invalidDatasourceConfiguration.setEndpoints(Collections.singletonList(endpoint));
 
         Assert.assertEquals(
-                Set.of("Missing port for endpoint", "Missing username for authentication.", "Missing password for authentication."),
+                Set.of("Missing username for authentication.", "Missing password for authentication."),
                 pluginExecutor.validateDatasource(invalidDatasourceConfiguration)
         );
     }

